mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -230,16 +239,16 @@ impl Deserializable for BlockHeader {
             None
         };
 
-        let is_zcash = (version == 4 && !reader.coin_variant().is_btc() && !reader.coin_variant().is_ppc())
+        let is_zcash_style = (version == 4 && !reader.coin_variant().is_btc() && !reader.coin_variant().is_ppc())
             || reader.coin_variant().is_kmd_assetchain();
-        let hash_final_sapling_root = if is_zcash { Some(reader.read()?) } else { None };
+        let mut hash_final_sapling_root = if is_zcash_style { Some(reader.read()?) } else { None };
         let time = reader.read()?;
-        let bits = if is_zcash {
+        let bits = if is_zcash_style {
             BlockHeaderBits::U32(reader.read()?)
```

Comment:
so is pivx a zcash_style or not?
- if it is, i see we are reading the hash_final_sapling_root before reading the nonce! (L244 & L251)
- if it is not, then we don't read the `solution` for it (L266)

Reply:
Also in the code there is this comment
```
// A PIVX header is a standard header with a `hash_final_sapling_root` added after the nonce.
```